memeolist/app/src/main/java/org/aerogear/android/app/memeolist/SyncService.java

```diff
@@ -0,0 +1,242 @@
+import org.aerogear.mobile.core.reactive.Requester;
+import org.aerogear.mobile.core.reactive.Responder;
+
+import java.io.File;
+import java.util.Map;
+
+import okhttp3.OkHttpClient;
+
+public final class SyncService {
+
+  public static final String TYPE = "sync";
+
+  private static SyncService instance;
+
+  private final ApolloClient apolloClient;
+
+  public SyncService(@Nonnull OkHttpClient okHttpClient, @Nonnull String serverUrl,
+                     @Nonnull String webSocketUrl, Context context) {
+    DiskLruHttpCacheStore cacheStore = createHttpCache();
+
+    ApolloClient.Builder builder = ApolloClient.builder();
+    builder.serverUrl(nonNull(serverUrl, "serverUrl"))
+            .okHttpClient(nonNull(okHttpClient, "okHttpClient"))
+            .httpCache(new ApolloHttpCache(cacheStore));
+            //.subscriptionTransportFactory(new Factory(webSocketUrl, okHttpClient));
+
+    NormalizedCacheFactory cacheFactory = new LruNormalizedCacheFactory(EvictionPolicy.builder().maxSizeBytes(10 * 1024).build());
+    //builder.normalizedCache(cacheFactory);
+    //createSQLCache(builder, context);
+
+    apolloClient = builder.build();
+  }
+
+  private void createSQLCache(ApolloClient.Builder builder, Context context) {
+    //Create the ApolloSqlHelper. Please note that if null is passed in as the name, you will get an in-memory SqlLite database that
+    // will not persist across restarts of the app.
+    ApolloSqlHelper apolloSqlHelper = ApolloSqlHelper.create(context, "memeo");
+
+    //Create NormalizedCacheFactory
+    NormalizedCacheFactory cacheFactory = new SqlNormalizedCacheFactory(apolloSqlHelper);
+
+    //Create the cache key resolver, this example works well when all types have globally unique ids.
+    CacheKeyResolver resolver = new CacheKeyResolver() {
+      @NotNull
+      @Override
+      public CacheKey fromFieldRecordSet(@NotNull ResponseField field, @NotNull Map<String, Object> recordSet) {
+        // Change to _id for server
+        return formatCacheKey((String) recordSet.get("id"));
+      }
+
+      @NotNull
+      @Override
+      public CacheKey fromFieldArguments(@NotNull ResponseField field, @NotNull Operation.Variables variables) {
+        return formatCacheKey((String) field.resolveArgument("id", variables));
+      }
+
+      private CacheKey formatCacheKey(String id) {
+        if (id == null || id.isEmpty()) {
+          return CacheKey.NO_KEY;
+        } else {
+          return CacheKey.from(id);
+        }
+      }
+    };
+    builder.normalizedCache(cacheFactory, resolver);
+  }
+
+  @NonNull
+  private DiskLruHttpCacheStore createHttpCache() {
+    //Directory where cached responses will be stored
+    File file = new File("/cache/");
+
+    //Size in bytes of the cache
+    int size = 1024 * 1024;
+
+    //Create the http response cache store
+    return new DiskLruHttpCacheStore(file, size);
+  }
+
+  public static SyncService getInstance(Context context) {
```

Comment:
`getInstance()` should not receive anything if so, I prefer move `getInstance()` to a `Builder`. 

Anyway, you don't need to pass a _Context_. `MobileCore` has an `ApplicationContext` inject there on initialization, you can fire `MobileCore.getInscante().getContext()`

Reply:
This is investigation PR that will be used to get improvements upstream. I'm not going to use any mobile core.